backend/src/main/java/com/google/sps/datastore/ParticipantDatastore.java

```diff
@@ -0,0 +1,150 @@
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Participant;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.Nullable;
+
+public final class ParticipantDatastore {
+
+  // Datastore Key/Property constants
+  private static final String KIND_PARTICIPANT = "Participant";
+  private static final String PROPERTY_USERNAME = "username";
+  private static final String PROPERTY_STARTTIMEAVAILABLE = "startTimeAvailable";
+  private static final String PROPERTY_ENDTIMEAVAILABLE = "endTimeAvailable";
+  private static final String PROPERTY_DURATION = "duration";
+  private static final String PROPERTY_CURRENTMATCHID = "currentMatchId";
+  private static final String PROPERTY_TIMESTAMP = "timestamp";
+
+  /** Formatter for converting between String and ZonedDateTime */
+  private static final DateTimeFormatter formatter = DateTimeFormatter.ISO_ZONED_DATE_TIME;
+
+  /** Datastore */
+  private final DatastoreService datastore;
+
+  /** Constructor */
+  public ParticipantDatastore(DatastoreService datastore) {
+    this.datastore = datastore;
+  }
+
+  /** Add Participant to datastore */
+  public void addParticipant(Participant participant) {
+    // Set properties of entity
+    Entity participantEntity = new Entity(KIND_PARTICIPANT, participant.getUsername());
+    participantEntity.setProperty(PROPERTY_USERNAME, participant.getUsername());
+    participantEntity.setProperty(
+        PROPERTY_STARTTIMEAVAILABLE, participant.getStartTimeAvailable().format(formatter));
+    participantEntity.setProperty(
+        PROPERTY_ENDTIMEAVAILABLE, participant.getEndTimeAvailable().format(formatter));
+    participantEntity.setProperty(PROPERTY_DURATION, participant.getDuration());
+    participantEntity.setProperty(PROPERTY_CURRENTMATCHID, participant.getCurrentMatchId());
+    participantEntity.setProperty(PROPERTY_TIMESTAMP, participant.getTimestamp());
+
+    // Insert entity into datastore
+    datastore.put(participantEntity);
+  }
+
+  /** Return Participant Entity from username */
+  private Entity getEntityFromUsername(String username) {
+    Key participantKey = KeyFactory.createKey(KIND_PARTICIPANT, username);
+    try {
+      return datastore.get(participantKey);
+    } catch (EntityNotFoundException e) {
+      // TODO: what error to send
+      return null;
+    }
+  }
+
+  /** Update Participant with new matchId, null out availability to show currently matched */
+  public void updateNewMatch(String username, long matchId) {
+    Entity participantEntity = getEntityFromUsername(username);
```

Comment:
Since `getEntityFromUsername` can return null on an EntityNotFoundException, what do you think about adding a null check here?